src/app_agent_websocket.rs

```diff
@@ -129,73 +142,78 @@ impl AppAgentWebsocket {
                     _ => None,
                 });
 
             let provisioned_cell = maybe_provisioned.ok_or(ConductorApiError::CellNotFound)?;
-            return Ok(provisioned_cell.cell_id);
+            Ok(provisioned_cell.cell_id)
         }
     }
 }
 
+pub enum ZomeCallTarget {
+    CellId(CellId),
+    /// You can call by role name for provisioned cells but any clone cells you create after
+    /// creating the [AppAgentWebsocket] will need to be called by [ZomeCallTarget::CellId].
+    RoleName(RoleName),
+    /// Any clone cells you create after creating the [AppAgentWebsocket] will need to be called by [ZomeCallTarget::CellId].
+    CloneId(CloneCellId),
+}
+
+impl From<CellId> for ZomeCallTarget {
+    fn from(cell_id: CellId) -> Self {
+        ZomeCallTarget::CellId(cell_id)
+    }
+}
+
+impl From<RoleName> for ZomeCallTarget {
+    fn from(role_name: RoleName) -> Self {
+        ZomeCallTarget::RoleName(role_name)
+    }
+}
+
+impl From<CloneCellId> for ZomeCallTarget {
+    fn from(clone_id: CloneCellId) -> Self {
+        ZomeCallTarget::CloneId(clone_id)
+    }
+}
+
 fn is_clone_id(role_name: &RoleName) -> bool {
-    role_name.as_str().contains(".")
+    role_name.as_str().contains('.')
 }
 
 fn get_base_role_name_from_clone_id(role_name: &RoleName) -> RoleName {
     RoleName::from(
         role_name
             .as_str()
-            .split(".")
-            .into_iter()
+            .split('.')
             .map(|s| s.to_string())
             .collect::<Vec<String>>()
             .first()
             .unwrap(),
     )
 }
 
-/// Signs an unsigned zome call with the given LairClient
-pub async fn sign_zome_call_with_client(
-    zome_call_unsigned: ZomeCallUnsigned,
-    client: &LairClient,
-) -> Result<ZomeCall, String> {
-    // sign the zome call
-    let pub_key = zome_call_unsigned.provenance.clone();
-    let mut pub_key_2 = [0; 32];
-    pub_key_2.copy_from_slice(pub_key.get_raw_32());
-
-    let data_to_sign = zome_call_unsigned
-        .data_to_sign()
-        .map_err(|e| format!("Failed to get data to sign from unsigned zome call: {}", e))?;
-
-    let sig = client
-        .sign_by_pub_key(pub_key_2.into(), None, data_to_sign)
-        .await
-        .map_err(|e| format!("Failed to sign zome call by pubkey: {}", e.str_kind()))?;
-
-    let signature = Signature(*sig.0);
-
-    let signed_zome_call = ZomeCall {
-        cell_id: zome_call_unsigned.cell_id,
-        zome_name: zome_call_unsigned.zome_name,
-        fn_name: zome_call_unsigned.fn_name,
-        payload: zome_call_unsigned.payload,
-        cap_secret: zome_call_unsigned.cap_secret,
-        provenance: zome_call_unsigned.provenance,
-        nonce: zome_call_unsigned.nonce,
-        expires_at: zome_call_unsigned.expires_at,
-        signature,
-    };
-
-    return Ok(signed_zome_call);
+/// Make the [AppWebsocket] functionality available through the [AppAgentWebsocket]
+impl Deref for AppAgentWebsocket {
```

Comment:
This feels nice. There is a slight oddity that you can't call `call_zome` on the `AppWebsocket` without explaining to Rust that you mean to do so, but I really don't see why you would when you have the simple version on this type already.